core/crypto/signed_hdr.c

```diff
@@ -44,29 +44,30 @@ struct shdr *shdr_alloc_and_copy(const struct shdr *img, size_t img_size)
 	return shdr;
 }
 
-TEE_Result shdr_verify_signature(const struct shdr *shdr)
+TEE_Result shdr_verify_signature(const struct shdr *shdr,
+				 struct ftmn_func_arg *ftmn_arg)
```

Comment:
Each function that need this handshake (`FTMN_CALL_FUNC()` / `FTMN_CALLEE_DONE()`) need this parameter. It could have been passed by some thread specific data instead, but that only hides that part of the API.